stratumv2-lib/src/frame.rs

```diff
@@ -0,0 +1,82 @@
+use crate::error::{Error, Result};
+use crate::parse::{serialize, ByteParser, Deserializable, Serializable};
+use crate::types::{MessageType, U24};
+use std::io;
+
+/// Used to deserialize a received network frame. The payload would be further
+/// deserialized according to the received MessageTypes.
+pub struct Message {
+    pub message_type: MessageType,
+    pub payload: Vec<u8>,
+}
+
+impl Message {
+    pub fn new<T: Into<Vec<u8>>>(message_type: MessageType, payload: T) -> Message {
+        Message {
+            message_type,
+            payload: payload.into(),
+        }
+    }
+}
+
+impl Serializable for Message {
+    fn serialize<W: io::Write>(&self, writer: &mut W) -> Result<usize> {
+        let message_type: u8 = self.message_type.msg_type();
+        let mut extension_type: u16 = self.message_type.ext_type();
+        if self.message_type.channel_bit() {
+            extension_type |= 0x8000;
+        }
+
+        let message_length = U24::new(self.payload.len() as u32)?;
+
+        let length = extension_type.serialize(writer)?
+            + message_type.serialize(writer)?
+            + message_length.serialize(writer)?
+            + writer.write(self.payload.as_slice())?;
+
+        Ok(length)
+    }
+}
+
+impl Deserializable for Message {
+    fn deserialize(parser: &mut ByteParser) -> Result<Message> {
+        let mut extension_type = u16::deserialize(parser)?;
+        let channel_bit = (extension_type & 0x8000) != 0;
+        extension_type &= 0x7FFF;
```

Comment:
I get the MSB check for the channel bit, I don't really get the purpose of `extenstion_type &= 0x7FFF`? Is it acting as a Bit Mask???